app/controllers/users_controller.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const bcrypt = require('bcryptjs'),
+  userService = require('../services/users'),
+  validationResult = require('express-validator/check').validationResult,
+  errors = require('../errors');
+
+exports.create = (req, res, next) => {
+  const saltRounds = 10;
+
+  const validationErrors = validationResult(req);
+  if (!validationErrors.isEmpty()) {
+    return next(errors.badRequest(validationErrors.mapped()));
+  }
+
+  const user = req.body
+    ? {
+        firstName: req.body.firstName,
+        lastName: req.body.lastName,
+        username: req.body.username,
+        password: req.body.password,
+        email: req.body.email
+      }
+    : {};
+
+  bcrypt
+    .hash(user.password, saltRounds)
+    .then(hash => {
+      user.password = hash;
+
+      userService
```

Comment:
return this promise, otherwise it wont reach the catch if an exception is thrown inside it